dspace-api/src/main/java/org/dspace/embargo/EmbargoServiceImpl.java

```diff
@@ -151,7 +151,7 @@ public void liftEmbargo(Context context, Item item)
         throws SQLException, AuthorizeException, IOException
     {
         // new version of Embargo policies remain in place.
-        //lifter.liftEmbargo(context, item);
+        lifter.liftEmbargo(context, item);
```

Comment:
This change looks to be unnecessary. This line of code was explicitly commented out when the new embargo system was added in DSpace 3.  The comment above (while badly worded) seems to note that traditional embargos get "translated" into new embargo system (as resource policy dates), so this explicit `liftEmbargo` shouldn't be necessary.

Here's the DSpace 3 commit which added new embargo (and commented out this line): https://github.com/DSpace/DSpace/commit/4b9b206d8c1455cc556688c726ae0ed596d25205#diff-b11c7ae8084a53137fcf0dca05b9fce6R179

All that said, if you are finding that traditional embargos are *not getting lifted properly in 6.x*, then that is definitely a bug.  But, since this line has been commented out since 3.x, I don't think it's the source of the bug (as traditional embargoes work fine in 5.x).

Reply:
My recommendation is that the embargo lifter should only remove the lift date metadata.  It should not attempt to modify the access policies for bitstreams.

Reply:
I have to admit I'm still a bit confused about this one. 

In our test server, legacy embargo was not working in DSpace 6 (at least with our data). By legacy embargo, I mean running the daily cron script dspace embargo-lifter (and I definitely assumed that lifting the embargo involves modifying the access policies for bitstreams). At this stage migrate-embargo utility was not used.

Then I found DS-2862 where lifter.liftEmbargo line was added (again) in DSpace 5. Based on this I assumed that traditional embargoes do not work correctly in 5.x either. But: based on @tdonohue :s comment I believe that there is a bug with lifting traditional embargoes. If the line  lifter.liftEmbargo is supposed to be commented, I suggest that the changes introduced in DSpace 3 commit are investigated w.r.t DSpace 6 code to fix the legacy embargo properly.

Other changes in this PR so far should not affect the lifting functionality itself, just reporting about reporting about Anonymous access to bitstreams (and I agree that this should yet be supplemented with policy date check).

Reply:
Hi @minurmin : It's my understanding that the `lifter.liftEmbargo()` line is no longer needed because traditional embargoes essentially get "translated" into new embargoes automatically by the `setEmbargo()` method. 

https://github.com/DSpace/DSpace/blob/dspace-6_x/dspace-api/src/main/java/org/dspace/embargo/DefaultEmbargoSetter.java#L88-L108

Notice how that method calls `getEmbargoTermsAsDate()`, which reads the embargo date set in embargo metadata fields:
https://github.com/DSpace/DSpace/blob/dspace-6_x/dspace-api/src/main/java/org/dspace/embargo/EmbargoServiceImpl.java#L113-L146

And, then `setEmbargo()` takes that date and creates Resource Policies on all bundles/bitstreams to essentially "transform" this into a new embargo.

That's why we believe the `liftEmbargo()` method should be unnecessary.  Essentially, a "traditional embargo" should be auto-translated into a "new embargo" at the time the item is saved into the archive, as `setEmbargo()` is called in the `InstallItemService.finishItem()` method which installs the item into the Archive: https://github.com/DSpace/DSpace/blob/dspace-6_x/dspace-api/src/main/java/org/dspace/content/InstallItemServiceImpl.java#L236

So, if this isn't working properly, we may want to more closely walk through / debug the code here. Based on the logic I see, `liftEmbargo()` should be unnecessary (and should remain commented out).  But, that may mean there's a bug somewhere else that we are missing.

Reply:
Ok, I think I finally got it. Many thanks for the clarification, it seems that I did not quite understand the meaning of "legacy embargo" in DSpace 6 context. 

Confirmed: if the item is submitted to DSpace 6 with an editable dc.date.embargo metadata field, default embargo setter sets the timed policy as expected.

However, if the database is migrated from a pre-3.0 DSpace, the timed policies are _not_ in place (i.e. an item with legacy embargo has no read policy on the bitstream). One has to run dspace migrate-embargo -a to generate the policies - which I didn't do since I assumed (based on description of DS-2588, e.g. "This type of embargo can only be lifted by running the ./dspace embargo-lifter command") that the legacy embargo would work in 6.0 as in 1.7 (=i.e. legacy embargo would involve no timed policies). So the "fix" would involve updating the documentation to stress out the fact that it is necessary to run  migrate-embargo if updating from pre 3.0 DSpace, and note that the intended behaviour of the legacy embargo lifter utility has been changed. At least:
* https://wiki.duraspace.org/display/DSDOC6x/Embargo#Embargo-Pre-3.0EmbargoMigrationRoutine (contains no explanation when the migration utility should be run)
* https://wiki.duraspace.org/display/DSDOC6x/Upgrading+DSpace (makes no mention about migrating embargoes)

Considering the immediate problem of getting embargoes to work with DSpace 6 I now agree that the line lifter.liftEmbargo is not needed after all. The basic functionality more or less works as expected, provided that one understands the intended usage. 

It should be noted that with post 3.0 code customized embargo lifters (e.g. https://groups.google.com/d/topic/dspace-tech/UpA9MfIqn2g/discussion) cannot be used. However, this too is probably more of a documentation issue (cf. https://jira.duraspace.org/browse/DS-2588?focusedCommentId=58931&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-58931 ) than a real problem since the lifter functionality can easily be reintroduced in a local fork.